internal/manifests/mutate.go

```diff
@@ -270,7 +274,9 @@ func mutateDaemonset(existing, desired *appsv1.DaemonSet) error {
 	if existing.CreationTimestamp.IsZero() {
 		existing.Spec.Selector = desired.Spec.Selector
 	}
-
+	if err := mergeWithOverwriteWithEmptyValue(&existing.Spec.Template.Spec.NodeSelector, desired.Spec.Template.Spec.NodeSelector); err != nil {
```

Comment:
shouldn't this step come after the following mergeWithOverride? Otherwise we would override this again in the next step?